cpp/arcticdb/version/local_versioned_engine.cpp

```diff
@@ -1160,31 +1160,31 @@ SpecificAndLatestVersionKeys LocalVersionedEngine::get_stream_index_map(
     return std::make_pair(specific_versions, latest_versions);
 }
 
+folly::Future<std::pair<std::optional<VariantKey>, std::optional<google::protobuf::Any>>> LocalVersionedEngine::get_metadata_future_impl(
```

Comment:
I believe we agreed to remove `future` from the method names. I think a reasonable convention to set would be to have `get_metadata_async` call `get_metadata`

Reply:
`read_metadata` currently has it's own (almost identical) implementation in `version_store_api.cpp`. Please replace that with a call to this function.

Reply:
agree 100%